tests/functional/context_methods/test_cli_vars.py

```diff
@@ -206,3 +206,38 @@ def test_vars_in_selectors(self, project):
         # Var in cli_vars works
         results = run_dbt(["run", "--vars", "snapshot_target: dev"])
         assert len(results) == 1
+
+
+models_scrubbing__schema_yml = """
+version: 2
+models:
+- name: simple_model
+  columns:
+  - name: simple
+    data_tests:
+    - accepted_values:
+        values:
+        - abc
+"""
+
+models_scrubbing__simple_model_sql = """
+select
+    '{{ var("DBT_ENV_SECRET_simple") }}'::varchar as simple
+"""
+
+
+class TestCLIVarsScrubbing:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "schema.yml": models_scrubbing__schema_yml,
+            "simple_model.sql": models_scrubbing__simple_model_sql,
+        }
+
+    def test__run_results_scrubbing(self, project):
+        results = run_dbt(["run", "--vars", "{DBT_ENV_SECRET_simple: abc, unused: def}"])
+        assert len(results) == 1
+        results = run_dbt(["test", "--vars", "{DBT_ENV_SECRET_simple: abc, unused: def}"])
+        assert len(results) == 1
```

Comment:
@nielspardon I looked at this along with @ChenyuLInx 

One question we have:
- Does it need to separate `run` and `test` here? Or would just `run` suffice? If it needs to be both `run` and `test`, then would `build` work instead?